tests.py

```diff
@@ -16,36 +16,26 @@
 import matplotlib
 matplotlib.use('agg')
 
-import nose
+# import nose
+import pytest
 from matplotlib import default_test_modules
 
 
 def run(extra_args):
-    from nose.plugins import multiprocess
-
-    env = matplotlib._get_nose_env()
-
     matplotlib._init_tests()
 
-    # Nose doesn't automatically instantiate all of the plugins in the
-    # child processes, so we have to provide the multiprocess plugin with
-    # a list.
-    plugins = matplotlib._get_extra_test_plugins()
-    multiprocess._instantiate_plugins = plugins
-
-    nose.main(addplugins=[x() for x in plugins],
-              defaultTest=default_test_modules,
-              argv=sys.argv + extra_args,
-              env=env)
+    argv = sys.argv + extra_args
+    # pytest.main(['--pyargs', '--cov=matplotlib'] + default_test_modules)
+    print(argv + ['--pyargs'] + default_test_modules)
+    pytest.main(argv + ['--pyargs'] + default_test_modules)
```

Comment:
I'd try calingl `sys.exit()` with the return value here so it's passed back to the os.